### PR TITLE
Replace commit hash with branch name in rebase params

### PR DESCRIPTION
**Pull request by PawelLipski**
image

### CODE REVIEW

**Review by mkondratek — APPROVED** [review]
Awesome

File: backendImpl/src/main/java/com/virtuslab/gitmachete/backend/impl/GitRebaseParameters.java [hunk 1]
```diff
@@ -10,17 +10,17 @@
 @Data
 public class GitRebaseParameters implements IGitRebaseParameters {
   private final IManagedBranchSnapshot currentBranch;
-  private final ICommitOfManagedBranch newBaseCommit;
+  private final IManagedBranchSnapshot newBaseBranch;
   private final ICommitOfManagedBranch forkPointCommit;
 
   @ToString.Include(name = "currentBranch")
-  private String getCurrentBranchBame() {
```
Comment (mkondratek) [inline backendImpl/src/main/java/com/virtuslab/gitmachete/backend/impl/GitRebaseParameters.java:17]: Say my bame :trollface:

File: frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseRebaseBranchOntoParentAction.java [hunk 2]
```diff
@@ -233,10 +233,10 @@ public void run(ProgressIndicator indicator) {
   private GitRebaseParams getIdeaRebaseParamsOf(GitRepository repository, IGitRebaseParameters gitRebaseParameters) {
     GitVersion gitVersion = repository.getVcs().getVersion();
     String currentBranch = gitRebaseParameters.getCurrentBranch().getName();
-    String newBase = gitRebaseParameters.getNewBaseCommit().getHash();
+    String newBaseBranch = gitRebaseParameters.getNewBaseBranch().getFullName();
```
Comment (micpiotrowski) [inline frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseRebaseBranchOntoParentAction.java:236]:
```suggestion
    String newBaseBranchFullName = gitRebaseParameters.getNewBaseBranch().getFullName();
```
?

**Review by micpiotrowski — APPROVED** [review]

File: frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseRebaseBranchOntoParentAction.java (hunk 2 above)
Comment (PawelLipski) [reply frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseRebaseBranchOntoParentAction.java:236]: Done

**Pull request merged**